### PR TITLE
[Core][MPI] Missing Flags in the partition of the ModelPartIO

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Missing Flags in the partition of the ModelPartIO

**🆕 Changelog**
- [Adding DivideFlagVariableData to ModelPartIO](https://github.com/KratosMultiphysics/Kratos/commit/800cf7a7465db5a9e8af403aaad695d28b1e1826)
- [Typo](https://github.com/KratosMultiphysics/Kratos/commit/54c8d4d4bab6c34eb968ddcae17479cbac6a98b2)
- [Style correction](https://github.com/KratosMultiphysics/Kratos/commit/10fe2e7443a0252763e8d81c033f3467be2a31f3)
- [Adding test for flag divide](https://github.com/KratosMultiphysics/Kratos/commit/29d1927a94e9a7e52440cd8edb621f247b78dbfe)

### CODE REVIEW

File: kratos/sources/model_part_io.cpp [hunk 1]
```diff
@@ -2053,13 +2054,9 @@ void ModelPartIO::ReadNodalDataBlock(ModelPart& rThisModelPart)
 
     VariablesList rThisVariables = rThisModelPart.GetNodalSolutionStepVariablesList();
 
-
-    if(KratosComponents<Flags >::Has(variable_name))
-    {
+    if(KratosComponents<Flags >::Has(variable_name)) {
```
Comment (roigcarlo) [inline kratos/sources/model_part_io.cpp:2057]: could yo move it at the Matrix, Vector, VariableData, etc...?

File: kratos/sources/model_part_io.cpp [hunk 2]
```diff
@@ -4152,47 +4129,30 @@ void ModelPartIO::DivideNodalDataBlock(OutputFilesContainerType& OutputFiles,
     WriteInAllFiles(OutputFiles, variable_name);
     WriteInAllFiles(OutputFiles, "\n");
 
-    if(KratosComponents<Variable<double> >::Has(variable_name))
-    {
+    if(KratosComponents<Flags>::Has(variable_name)) {
```
Comment (roigcarlo) [inline kratos/sources/model_part_io.cpp:4132]: Same

File: kratos/sources/model_part_io.cpp (hunk 1 above)
Comment (loumalouomega) [reply kratos/sources/model_part_io.cpp:2057]: I don't understand

File: kratos/sources/model_part_io.cpp (hunk 2 above)
Comment (loumalouomega) [reply kratos/sources/model_part_io.cpp:4132]: Same, i don't understand

File: kratos/sources/model_part_io.cpp (hunk 1 above)
Comment (roigcarlo) [reply kratos/sources/model_part_io.cpp:2057]: I would expect the "simple" variables at the begining of the if, and the "complex" at the end, its like that all over Kratos (int,double,arrays,vector,matrices,etc...)

Comment (loumalouomega) [reply kratos/sources/model_part_io.cpp:2057]: But the flags was already there, take a look, in here I am  just correcting the {} style

Comment (loumalouomega) [reply kratos/sources/model_part_io.cpp:2057]: And Flags are IMHO the simplest ones

**Review by roigcarlo — APPROVED** [review]
ok

**Pull request merged**